gallery/tutorials/plot_lecture_feature_demo.py

```diff
@@ -0,0 +1,460 @@
+)
+
+# Estimate the white noise.
+noise_estimator = WhiteNoiseEstimator(sim_wht)
+
+# %%
+# A Custom ``FunctionFilter``
+# ---------------------------
+#
+# We will now apply some more interesting noise, using a custom function, and then apply a ``whitening`` process to our data.
+#
+# Using ``FunctionFilter`` we can create our own custom functions to apply in a pipeline.
+# Here we want to apply a custom filter as a noise adder.  We can use a function of two variables for example.
+
+
+def noise_function(x, y):
+    return 1e-7 * np.exp(-(x * x + y * y) / (2 * 0.3 ** 2))
+
+
+# In python, functions are first class objects.
+# We take advantage of that to pass this function around as a variable.
+# It will be evaluated later...
+custom_noise_filter = FunctionFilter(noise_function)
+
+# Create yet another Simulation source to tinker with.
+sim4 = Simulation(
+    L=v2.resolution, n=num_imgs, vols=v2, noise_filter=custom_noise_filter
+)
+sim4.images(0, 10).show()
+
+# %%
+# Noise Whitening
+# ---------------
+#
+# Applying the ``Simulation.whiten()`` method just requires passing the filter corresponding to the estimated noise instance.
+# Then we can inspect some of the whitened images.  While noise is still present, we can see a dramatic change.
+
+# Estimate noise.
+aiso_noise_estimator = AnisotropicNoiseEstimator(sim4)
+
+# Whiten based on the estimated noise
+sim4.whiten(aiso_noise_estimator.filter)
+
+# What do the whitened images look like...
+sim4.images(0, 10).show()
+
+# %%
+# Homework Task 4
+# ^^^^^^^^^^^^^^^
+#
+# Try some other image preprocessing methods exposed by the ``Simulation``/``ImageSource`` classes.
+#
+# Try some other custom function to add noise or other corruptions to the images.
+
+# %%
+# Real Experimental Data - ``RelionSource``
+# -----------------------------------------
+#
+# Now that we know our experiment code seems to run,
+# we can try to replace the simulation with a real experimental data source.
+#
+# Lets attempt the same CL experiment, but with a ``RelionSource``.
+
+src = RelionSource(
+    "../../tests/saved_test_data/sample_relion_data.star",
+    data_folder="",
+    pixel_size=5.0,
+    max_rows=1024,
+)
+
+# Data resides on Tiger Cluster
+# Please make sure you are using a compute node once you've installed ASPIRE, not the head node...
+# src =  RelionSource(
+#    "/tigress/gbwright/data/cryo-em/CryoEMdata/empiar10028/shiny_2sets.star", data_folder="", pixel_size=5.0, max_rows=100
+# )
+src.downsample(img_size)
+
+src.images(0, 10).show()
+
+noise_estimator = WhiteNoiseEstimator(src)
+src.whiten(noise_estimator.filter)
+
+orient_est = CLSyncVoting(src, n_theta=36)
+orient_est.estimate_rotations()
+rots_est = orient_est.rotations
+
+# %%
+# We can see that the code can easily run with experimental data by subsituting the ``Source`` class.
+# However, we have hit the practical limitation that requires class averaging of images....
```

Comment:
Can we add an issue here to include class averaging in the script, which would complete this part?